examples/ant_world/ant_world.cc

```diff
@@ -68,41 +52,25 @@ int main(int argc, char **argv)
     // Whether to use the 3D reconstructed Rothamsted model
     const bool useRothamstedModel = argc > 1 && strcmp(argv[1], "--rothamsted") == 0;
 
-    // Set GLFW error callback
-    glfwSetErrorCallback(handleGLFWError);
-
-    // Initialize the library
-    if(!glfwInit()) {
-        LOGE << "Failed to initialize GLFW";
-        return EXIT_FAILURE;
-    }
-
-    // Prevent window being resized
-    glfwWindowHint(GLFW_RESIZABLE, false);
+    sf::ContextSettings settings;
+    settings.antialiasingLevel = 8;
 
-    // Create a windowed mode window and its OpenGL context
-    GLFWwindow *window = glfwCreateWindow(width, height, "Ant world", nullptr, nullptr);
-    if(!window)
-    {
-        glfwTerminate();
-        LOGE << "Failed to create window";
-        return EXIT_FAILURE;
-    }
+    // Create SFML window
+    sf::Window window(sf::VideoMode{ width, height },
+                      "Ant world example",
+                      sf::Style::Titlebar | sf::Style::Close,
+                      settings);
 
-    // Make the window's context current
-    glfwMakeContextCurrent(window);
+    // Enable VSync
+    window.setVerticalSyncEnabled(true);
+    window.setActive(true);
 
     // Initialize GLEW
     if(glewInit() != GLEW_OK) {
         LOGE << "Failed to initialize GLEW";
         return EXIT_FAILURE;
     }
 
-    // Enable VSync
-    glfwSwapInterval(1);
-
-    glDebugMessageCallback(handleGLError, nullptr);
```

Comment:
Does SFML do this internally and convert GL errors to exceptions? If not, without this second call, you have to use OpenGL in gross "wrapping each function call in a macro and checking it's return" mode

Reply:
I assumed it did, but it seems not: https://www.sfml-dev.org/tutorials/2.5/window-opengl.php

I'll reinstate this.